src/category_theory/presheaf.lean

```diff
@@ -0,0 +1,118 @@
+-- Copyright (c) 2018 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Scott Morrison, Mario Carneiro, Reid Barton
+
+import category_theory.instances.topological_spaces
+import category_theory.whiskering
+import category_theory.natural_isomorphism
+import topology.basic
+
+open topological_space
+
+universes v u
+
+open category_theory
+open category_theory.instances
+
+namespace category_theory
+
+variables (C : Type u) [𝒞 : category.{v+1} C]
+include 𝒞
+
+def presheaf (X : Top.{v}) := opens X ⥤ C
```

Comment:
I'm not sure whether you even really need this definition, but the name `category.presheaf` should refer to
```lean
def presheaf (C : Type v) [𝒞 : small_category.{v} C] := Cᵒᵖ ⥤ Type v
```
(or maybe some generalization to arbitrary universes). Also this whole file is specialized to topological spaces, so the module name `category_theory.presheaf` doesn't seem appropriate.

Reply:
`presheaf_on_space`? It think it warrants a name. (In particular, I like have the projection notation available to write `ℱ.pushforward f`, although I guess this is backwards, and the `f` should come first according to usual notation.)

"presheaf" gets used in two ways: `Cᵒᵖ ⥤ Type v`, or as `(opens X)ᵒᵖ ⥤ D`. The common generalisation is just a covariant functor, so it's hard to know how to name things.

Reply:
I've gone with `presheaf_on_space` for now.